backend/src/test/java/ai/verta/modeldb/TestsInit.java

```diff
@@ -165,15 +166,44 @@ public static void setServerAndService() throws Exception {
   }
 
   @AfterClass
-  public static void removeServerAndService() {
+  public static void removeServerAndService() throws InterruptedException {
     App.initiateShutdown(0);
 
+    cleanUpResources();
+
+    // shutdown test server
+    serverBuilder.build().shutdownNow();
+  }
+
+  protected static void cleanUpResources() throws InterruptedException {
     // Remove all entities
     // removeEntities();
     // Delete entities by cron job
-    deleteEntitiesCron.run();
+    SoftDeleteProjects softDeleteProjects = ReconcilerInitializer.softDeleteProjects;
+    SoftDeleteExperiments softDeleteExperiments = ReconcilerInitializer.softDeleteExperiments;
+    SoftDeleteExperimentRuns softDeleteExperimentRuns =
+        ReconcilerInitializer.softDeleteExperimentRuns;
+
+    softDeleteProjects.resync();
+    while (softDeleteProjects.isElementsNotDeleted()) {
+      LOGGER.trace("Project deletion is still in progress");
+    }
```

Comment:
Never do a kind of loop like this without a sleep. This will go crazy consuming CPU for no reason. Even sleeping for 10ms will make this better.

Reply:
Fixed